src/mpl/src/shm/mpl_shm_mmap.c

```diff
@@ -47,7 +47,8 @@ static inline int MPL_shm_seg_create_attach_templ(MPL_shm_hnd_t hnd, intptr_t se
                                                   void **shm_addr_ptr, int offset, int flag,
                                                   MPL_memory_class class)
 {
-    MPLI_shm_lhnd_t lhnd = -1, rc = -1;
+    MPLI_shm_lhnd_t lhnd = -1;
+    int rc = -1, failure = 0;
```

Comment:
Rather than use a `failure` variable, can we add an `fn_fail` label and set `rc` to -1 there? That is more closely aligned with other parts of MPICH/MPL.

Reply:
We need a second variable `failure` because we want to close the handle even in the case of an error, but return `-1` and not the output of `rc = MPLI_shm_lhnd_close(hnd);`.

Reply:
Ah, I see. Well this is still an improvement so I suppose we shouldn't block on it. @minsii do you have any comments? The other patches look fine to me.

Reply:
Actually I prefer the code style suggested by Ken as it is consistent with other MPICH code. Instead of the `failure` branch, we could do:
```
int rc = 0, rc_close = 0; /* init */
...
if (error returned)
    goto fn_fail;

fn_exit:
    if (MPLI_shm_lhnd_is_valid(hnd)) {
        rc_close = MPLI_shm_lhnd_close(hnd);
    }
    return (rc == -1) ? rc : rc_close;
fn_fail:
    rc = -1;
    goto fn_exit;
```

Reply:
I made this change